### PR TITLE
Add jobs

### PR DESCRIPTION
**Pull request by ThouCheese**

### CODE REVIEW

File: blockjoy/v1/node.proto [hunk 1]
```diff
@@ -370,11 +372,35 @@ message NodeProperty {
...
+  JOB_STATUS_FINISHED = 3;
+}
+
+// This message indicates the amount of progress a job has made towards its
+// completion. Note that it is possible for a job to report an amount of
+// progress without it also providing a total.
+message JobProgress {
+  // The total units of progress that need to be made in order to complete this
+  // job.
+  optional uint32 total = 1;
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:400]: After a while, I think there is no reason, to make it optional.

File: blockjoy/v1/node.proto [hunk 2]
```diff
@@ -370,11 +372,35 @@ message NodeProperty {
...
+message Job {
+  string name = 1;
+  JobStatus status = 2;
+  repeated string logs = 3;
+  uint64 restarts = 4;
+  optional JobProgress progress = 5;
+}
+
+// Flags whether a job is in an ok, failed or finished state.
+enum JobStatus {
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:387]: Lets keep it consistent with BabelAPI:
```
pub enum JobStatus {
    /// The current job was requested to start, but the process has not been launched yet.
    /// It was not picked by the JobRunner yet, or needs another job to be finished first.
    /// Every job starts with this state.
    Pending,
    /// The JobRunner actually picked that job.
    Running,
    /// Job finished - successfully or not. It means that the JobRunner won't try to restart that job anymore.
    Finished {
        /// Job exit code, if any. For `run_sh` job type it is script exit code.
        /// `None` always means some error, usually before the process was even started (e.g. needed job failed).  
        exit_code: Option<i32>,
        /// Error description or empty if successful.
        message: String,
    },
    /// Job was explicitly stopped.
    Stopped,
}
```

File: blockjoy/v1/node.proto (hunk 1 above)
Comment (ThouCheese) [reply blockjoy/v1/node.proto:416]: I will leave it optional nonetheless for forwards compatibility, I see no reason to make it required

File: blockjoy/v1/node.proto (hunk 2 above)
Comment (ThouCheese) [reply blockjoy/v1/node.proto]: Alright updated!

File: blockjoy/v1/node.proto [hunk 3]
```diff
@@ -376,25 +376,47 @@ message NodeProperty {
...
+  // A displayable message that the user might be able to read in order to
+  // further identify the current status of the job. For example, if the job has
+  // failed, an error message might be place here.
+  optional string message = 4;
+  // A list of log lines with information about the current job.
+  repeated string logs = 5;
+  // This field contains the number of restarts that as happened since the last
+  // cut-off. This is _not_ guaranteed to be the number of restarts that has
+  // happened since the creation of this node. Rather, this number may be reset
+  // to zero periodically, or whenever the job triggers the state `OK`.
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:394]: There is no state `OK` anymore.
```suggestion
  // to zero periodically, or whenever the job triggers the state `FINISHED`.
```

File: blockjoy/v1/node.proto (hunk 1 above)
Comment (mjr-blockjoy) [reply blockjoy/v1/node.proto:422]: Just thought it doesn't make sense if whole `NodeJobProgress progress = 7;` is already optional, but nvm.

File: blockjoy/v1/node.proto [hunk 4]
```diff
@@ -376,25 +376,47 @@ message NodeProperty {
...
+  uint64 restarts = 6;
+  // This message contains a couple of fields that together denote the progress
+  // towards completion of this job.
+  optional NodeJobProgress progress = 7;
 }
 
 // Flags whether a job is in an ok, failed or finished state.
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:401]:
```suggestion
// Flags describing a job possible states.
```

**Review by mjr-blockjoy — APPROVED** [review]

File: blockjoy/v1/node.proto [hunk 5]
```diff
@@ -370,11 +372,57 @@ message NodeProperty {
...
+  // The job has not started yet.
+  JOB_STATUS_PENDING = 1;
+  // The job is current being executed.
+  JOB_STATUS_RUNNING = 2;
+  // The job has successfully finished.
+  JOB_STATUS_FINISHED = 3;
+  // The job has unsuccessfully finished.
+  JOB_STATUS_FAILED = 4;
+  // The job was interrupted.
+  JOB_STATUS_STOPPED = 5;
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:413]: Linter 
```suggestion
  NODE_JOB_STATUS_UNSPECIFIED = 0;
  // The job has not started yet.
  NODE_JOB_STATUS_PENDING = 1;
  // The job is current being executed.
  NODE_JOB_STATUS_RUNNING = 2;
  // The job has successfully finished.
  NODE_JOB_STATUS_FINISHED = 3;
  // The job has unsuccessfully finished.
  NODE_JOB_STATUS_FAILED = 4;
  // The job was interrupted.
  NODE_JOB_STATUS_STOPPED = 5;
```

File: blockjoy/v1/node.proto [hunk 6]
```diff
@@ -370,11 +372,57 @@ message NodeProperty {
...
+// A job is a process running on the host that is necessary for the operation of
+// a node. Examples here are downloading the chain data or running the
+// blockchain node process.
+message Job {
+  // A name to identify this job by.
+  string name = 1;
+  // The status this job is currently in.
+  NodeJobStatus status = 2;
+  // If the process exited, then this field holds the unix-style exit code.
+  optional uint32 exit_code = 3;
```
Comment (mjr-blockjoy) [inline blockjoy/v1/node.proto:384]: Let's use signed int
```suggestion
  optional int32 exit_code = 3;
```

**Pull request merged**